explorer/testdata/operators/div_builtin.carbon

```diff
@@ -0,0 +1,19 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// RUN: %{explorer} %s 2>&1 | \
+// RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes=false %s
+// RUN: %{explorer} --parser_debug --trace_file=- %s 2>&1 | \
+// RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes %s
+// AUTOUPDATE: %{explorer} %s
+// CHECK: result: 2
+
+package ExplorerTest api;
+
+// TODO: T:! Div
+fn DoDiv[T:! DivWith(.Self) where .Result == .Self](x: T, y: T) -> T { return x / y; }
+
+fn Main() -> i32 {
+  return DoDiv(8, 3);
```

Comment:
```suggestion
fn Main() -> i32 {
  return 8 / 3;
```

Generally, this change looks great. I do want to ask about this though: is there a reason you aren't doing the division directly here, and instead have the function pass-through?

(giving the suggestion to show what I mean)

Reply:
I'm making it consistent with the tests for the other operations added in 2ce5867f80bbf2e750598e3b5eecfe5a00ef3df0 by @zygoloid. I don't have a reason to do so other than the consistency. @zygoloid WDYT?

Reply:
I've sent #2109 to remove the indirection, AFAICT it's already not in other operator tests such as mod, but maybe I'm misunderstanding why it's there.

Reply:
I removed indirection in my test.

Reply:
We should have a test that checks `DivWith` is properly implemented for `i32` (rather than the `/` operator just being special-cased in explorer, which would be enough to make this test pass).